src/ImageSharp/Common/Helpers/DenseMatrixUtils.cs

```diff
@@ -99,7 +101,7 @@ internal static class DenseMatrixUtils
 
             for (int i = 0; i < kernelLength; i++)
             {
-                int offsetY = (row + i - radiusY).Clamp(0, maxRow);
+                int offsetY = (row + i - radiusY).Clamp(minRow, maxRow);
```

Comment:
Ah! Bad James!

Reply:
I actually realized this needed to be fixed in the second complex 1D convolution pass as well, done in https://github.com/SixLabors/ImageSharp/pull/842/commits/15044d740319ca0ba702191d5efdb8367299d159.